src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,109 @@
+ */
+public class AddCommand extends Command {
+
+    /**
+     * Constructor for AddCommand, for ToDo task
+     *
+     * @param command ToDo command
+     * @param description Task name of ToDo
+     */
+    public AddCommand(String command, String description) {
+        super.command = command;
+        super.description = description;
+        super.date = "";
+    }
+
+    /**
+     * Overloaded constructor for AddCommand, for Event and Deadline tasks
+     *
+     * @param command Event or Deadline command
+     * @param description Task name of Event or Deadline
+     */
+    public AddCommand(String command, String description, String date) {
+        super.command = command;
+        super.description = description;
+        super.date = date;
+    }
+
+    private void todoProcess(String taskName, TaskList tasks) {
+        Task task = new ToDo(taskName);
+        tasks.add(task);
+        output = "Got it. I've added this task: \n"
+                + task.toString() + getRemainingTasks(tasks);
+    }
+
+    private void eventProcess(String description, String date, TaskList tasks) throws DukeException {
+        if (!Utility.isValidDate(date)) {
+            throw new DukeException(DukeExceptionType.INVALID_DATE_FORMAT);
+        }
+        Task task = new Event(description, LocalDate.parse(date));
+        tasks.add(task);
+        output = "Got it. I've added this task: \n"
+                + task.toString() + getRemainingTasks(tasks);
+    }
+
+    private void deadlineProcess(String description, String date, TaskList tasks) throws DukeException {
+        if (!Utility.isValidDate(date)) {
+            throw new DukeException(DukeExceptionType.INVALID_DATE_FORMAT);
+        }
+        Task task = new Deadline(description, LocalDate.parse(date));
+        tasks.add(task);
+        output = "Got it. I've added this task: \n"
+                + task.toString() + getRemainingTasks(tasks);
+    }
+
+    private String getRemainingTasks(TaskList tasks) {
+        return "\nNow you have " + tasks.size() + " tasks in the list.";
+    }
+
+    /**
+     * Adds task to TaskList, saves to storage file and outputs response to terminal
+     *
+     * @param tasks TaskList
+     * @param storage Storage instance
+     * @throws DukeException If invalid date given for event or deadline task
+     */
+    @Override
+    public void execute(TaskList tasks, Storage storage) throws DukeException {
+        switch (command) {
+        case "todo":
+            todoProcess(description, tasks);
+            break;
+        case "event":
+            eventProcess(description, date, tasks);
+            break;
+        case "deadline":
+            deadlineProcess(description, date, tasks);
+            break;
+        default:
+            break;
+        }
+        storage.save(tasks);
+    }
+
+    /**
+     * Determines if Exit is called by user
+     *
+     * @return false
+     */
+    @Override
+    public boolean isExit() {
```

Comment:
Although this does follow the naming to sound like a boolean, but just a minor issue that isExit as phrase does not seem right to me

Reply:
Hello, thank you for reviewing my code! I'll work on it :)